src/java.base/share/legal/siphash.md

```diff
@@ -4,12 +4,12 @@
 SipHash reference C implementation
 
 ```
-   Copyright (c) 2012-2021 Jean-Philippe Aumasson
-   <jeanphilippe.aumasson@gmail.com>
-   Copyright (c) 2012-2014 Daniel J. Bernstein <djb@cr.yp.to>
```

Comment:
Why would you remove a author or year range?

Reply:
To match the license claim in the code we are using:
https://github.com/openjdk/jdk/blob/fb8f2a0a929ebe7f65c69741712b89bbb403ade9/src/hotspot/share/classfile/altHashing.cpp#L32-L43

Reply:
But that isn't the code we're based on.  As noted in the comment here:
https://github.com/openjdk/jdk/blob/fb8f2a0a929ebe7f65c69741712b89bbb403ade9/src/hotspot/share/classfile/altHashing.cpp#L25-L29
the code in HotSpot is derived from this:
https://github.com/veorq/SipHash/blob/master/halfsiphash.c

Reply:
That header only contains a single function declaration for an entry point into the implementation.
HotSpot doesn't use that function, and doesn't have anything with a corresponding signature.  So it's
not in any way derived from that header.  The HotSpot code is derived from the .c file only, so that's
the license we should be referencing.

Reply:
The problem is that the project is a joined work and has multiple variants of copyrights (see the reader). I don’t think it’s on the safe side to pick a single (non recent) copyright - especially if you change attribution after the fact. But that’s just me

Reply:
Copyright and license are two different things. The project has a dual license which we attribute. Code in different files can have different copyrights (as is the case in e.g. our own OpenJDK code). We copied the copyright from the file that we looked at. If you feel it has the wrong copyright please contact the authors of that file and ask them to update their code.

Reply:
The file has a different license also. Either you use the joined copyright and dual license or only the single copyright. But I asked DJB to clarify.